### PR TITLE
Add setting suspicious type for low cardinality (fixed #4965)

### LINKED ISSUE #4965: Add `allow_suspicious_low_cardinality_types` setting.
If this setting is turned off (default), don't allow to create a table with LowCardinality of integers, enums, Nullable integers, etc... (all types that have fixed size in memory and the size is 8 or less). Always allow to attach table.

### PR DESCRIPTION
**Pull request by stavrolia**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

For changelog. Remove if this is non-significant change.

Category (leave one):
- Improvement

Short description (up to few sentences):
Add setting forbidden by default to create table with suspicious types for low cardinality

### CODE REVIEW

File: dbms/src/Interpreters/InterpreterCreateQuery.cpp [hunk 1]
```diff
@@ -523,6 +526,22 @@ BlockIO InterpreterCreateQuery::createTable(ASTCreateQuery & create)
...
 
+    /// Check low cardinality types in creating table if it was not cancelled in setting
+    if (!create.attach && !context.getSettingsRef().allow_suspicious_low_cardinality_types)
+    {
+        for (const auto & name_and_type_pair : columns.getAllPhysical())
+        {
+            if (const auto * current_type_ptr = typeid_cast<const DataTypeLowCardinality *>(name_and_type_pair.type.get()))
+            {
+
+                if (!isStringOrFixedString(current_type_ptr->getDictionaryType()) && !isStringOrFixedString(*removeNullable(current_type_ptr->getDictionaryType())))
```
Comment (alexey-milovidov) [inline dbms/src/Interpreters/InterpreterCreateQuery.cpp]: Redundand condition. You can always use removeNullable.

File: dbms/src/Interpreters/InterpreterCreateQuery.cpp [hunk 2]
```diff
@@ -523,6 +526,22 @@ BlockIO InterpreterCreateQuery::createTable(ASTCreateQuery & create)
...
+                    throw Exception("The type " + current_type_ptr->getName() + "   "  + " is not allowed for making low cardinality in creating table by default, because it could increase time of join and memory consumption. "
```
[the remark this answers is not in the document]
Comment (alexey-milovidov) [reply dbms/src/Interpreters/InterpreterCreateQuery.cpp]: Error message in not enough clean and explanatory.

[the remark this answers is not in the document]
Comment (stavrolia) [reply dbms/src/Interpreters/InterpreterCreateQuery.cpp]: How I can make it better?

File: dbms/src/Interpreters/InterpreterCreateQuery.cpp [hunk 3]
```diff
@@ -523,6 +526,21 @@ BlockIO InterpreterCreateQuery::createTable(ASTCreateQuery & create)
...
 
+    /// Check low cardinality types in creating table if it was not cancelled in setting
+    if (!create.attach && !context.getSettingsRef().allow_suspicious_low_cardinality_types)
+    {
+        for (const auto & name_and_type_pair : columns.getAllPhysical())
+        {
+            if (const auto * current_type_ptr = typeid_cast<const DataTypeLowCardinality *>(name_and_type_pair.type.get()))
+            {
+                if (!isStringOrFixedString(*removeNullable(current_type_ptr->getDictionaryType())))
+                    throw Exception("Using type " + current_type_ptr->getName() + " can negatively affect performance. Low cardinality is usefull only for types String, Nullable(String), FixedString and Nullable(FixedString). "
```
Comment (alexey-milovidov) [inline dbms/src/Interpreters/InterpreterCreateQuery.cpp]: useful - single l.

PS. Ask @Akazz about the proper wording.

**Pull request merged**